holoviews/core/spaces.py

```diff
@@ -513,6 +515,25 @@ def __call__(self, *args, **kwargs):
         return ret
 
 
+
+class Generator(Callable):
+    """
+    Generators are considered a special case of Callable that accept no
+    arguments and never memoize.
+    """
+
+    callable = param.Parameter(default=None, constant=True, doc="""
+         The generator function being wrapped.""")
+
+    @property
+    def argspec(self):
+        from inspect import ArgSpec
+        return ArgSpec(args=[], varargs=None, keywords=None, defaults=None)
+
+    def __call__(self):
+        return next(self.callable)
```

Comment:
No as the ``Counter`` stream has a single stream parameter called ``counter``. That would become an argument so wouldn't apply to argument-free functions/generators.

You *could* imagine Streams without parameters but attributes. E.g a type of ``Counter`` that tracks how many times it is triggered. Not sure why we would want that though.